MPF.Core/Modules/PS3CFW/Parameters.cs

```diff
@@ -0,0 +1,308 @@
+                    info.Artifacts["discpic"] = GetBase64(GetFullFile(getKeyBasePath + ".disc.pic", binary: true)) ?? string.Empty;
+                if (File.Exists(getKeyBasePath + ".getkey.log"))
+                    info.Artifacts["getkeylog"] = GetBase64(GetFullFile(getKeyBasePath + ".getkey.log")) ?? string.Empty;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override List<string> GetLogFilePaths(string basePath)
+        {
+            var logFiles = new List<string>();
+            string? getKeyBasePath = GetCFWBasePath(basePath);
+
+            if (this.System != RedumpSystem.SonyPlayStation3)
+                return logFiles;
+
+            switch (this.Type)
+            {
+                case MediaType.BluRay:
+                    if (File.Exists($"{getKeyBasePath}.getkey.log"))
+                        logFiles.Add($"{getKeyBasePath}.getkey.log");
+                    if (File.Exists($"{getKeyBasePath}.disc.pic"))
+                        logFiles.Add($"{getKeyBasePath}.disc.pic");
+
+                    break;
+            }
+
+            return logFiles;
+        }
+
+        #endregion
+
+        #region Information Extraction Methods
+
+        /// <summary>
+        /// Get a formatted datfile from the PS3 CFW output, if possible
+        /// </summary>
+        /// <param name="iso">Path to ISO file</param>
+        /// <returns></returns>
+        private static Datafile? GeneratePS3CFWDatafile(string iso)
+        {
+            // If the ISO file doesn't exist, we can't get info from it
+            if (!File.Exists(iso))
+                return null;
+
+            try
+            {
+                if (Hashing.Hasher.GetFileHashes(iso, out long size, out string? crc, out string? md5, out string? sha1))
+                {
+                    return new Datafile
+                    {
+                        Games =
+                        [
+                            new()
+                            {
+                                Roms =
+                                [
+                                    new Rom { Name = Path.GetFileName(iso), Size = size.ToString(), Crc = crc, Md5 = md5, Sha1 = sha1 },
+                                ]
+                            }
+                        ]
+                    };
+                }
+                return null;
+            }
+            catch
+            {
+                // We don't care what the exception is right now
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Get a isobuster-formatted PVD from a PS3 ISO, if possible
+        /// </summary>
+        /// <param name="isoPath">Path to ISO file</param>
+        /// <param name="pvd">Formatted PVD string, otherwise null</param>
+        /// <returns>True if PVD was successfully parsed, otherwise false</returns>
+        private static bool GetPVD(string isoPath, out string? pvd)
```

Comment:
This entire method feels like a good fit for `BaseParameters` or even in one of the places like `InfoTool`. It definitely could be used as a backup for other implementations. If it does move, it may be a good thing to split this into getting the PVD bytes and then formatting the PVD. Personal opinion on the second, of course.

Reply:
In its current form, I'm not sure I trust it being used for anything other than PS3 ISOs. Does PVD *always* start at 0x8320?
A proper base implementation would look for the magic string "CD001" and go from there, rather than hard coding the position. But, this naive implementation works for PS3 :)

Reply:
Assuming a 2048-sector ISO, I assume so, but sure, for now this can stay as its own single method.